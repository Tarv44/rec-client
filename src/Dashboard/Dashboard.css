.dashboard-header {
    margin-bottom: 20px;
}

.dashboard-section {
    background-color: #454545;
    display: flex;
    flex-direction: column;
}

.dashboard-section h2 {
    color:  white;
    display: inline-block;
    border-bottom: solid #F0E118 .5px;
    padding: .2em 0;
    width: auto;
    align-self: center;
    margin: 0;
}

.dashboard-section a {
    color: white;
    text-decoration: underline #0CF0AB;
}

.dashboard-section li {
    margin: .5em 0;
    font-size: 20px;
}



.dashboard-title {
    color: #454545;
    border-bottom: solid 3px #0CF0AB;
}