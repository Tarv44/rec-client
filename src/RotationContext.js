import React from 'react';

const RotationContext = React.createContext({
    updateUser: () => {}
})

export default RotationContext;