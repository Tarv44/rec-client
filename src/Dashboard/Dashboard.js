import { Component } from 'react';
import RotationContext from '../RotationContext';
import RecentEx from '../RecentEx/RecentEx';
import AllEx from '../AllEx/AllEx'
import './Dashboard.css';

export default class Dashboard extends Component {
    static contextType = RotationContext
    
    render() {
        return (
            <main>
                <header>
                    <h1>{this.context.user.username}</h1>
                </header>

                <RecentEx exchanges={this.context.exchanges}/>

                <AllEx exchanges={this.context.exchanges}/>
            </main>
        )
    }
}