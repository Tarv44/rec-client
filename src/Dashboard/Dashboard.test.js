import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Dashboard from './Dashboard'
import store from '../dummy-store'
import { BrowserRouter } from 'react-router-dom'
import RotationContext from '../RotationContext'


describe('----- Dashboard -----', () => {

    const context = {
        current_user: {
            id: 1,
            username: 'user'
        },
        current_exchanges: store.exchanges
    }
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <RotationContext.Provider value={context}>
                    <Dashboard />
                </RotationContext.Provider>
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <RotationContext.Provider value={context}>
                        <Dashboard />
                    </RotationContext.Provider>
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})