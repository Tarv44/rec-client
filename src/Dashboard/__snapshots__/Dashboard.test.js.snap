// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Dashboard ----- renders the UI as expected 1`] = `
<main
  className="dashboard"
>
  <header
    className="dashboard-header"
  >
    <h1
      className="dashboard-title"
    >
      user
    </h1>
    <a
      aria-current={null}
      href="/add-exchange"
      onClick={[Function]}
    >
      <button>
        Create New Exchange
      </button>
    </a>
  </header>
  <section
    className="dashboard-section recent"
  >
    <h2>
      Recently Updated
    </h2>
    <ul>
      <li>
        <a
          aria-current={null}
          href="/exchange/1"
          onClick={[Function]}
        >
          The Roots
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/exchange/2"
          onClick={[Function]}
        >
          Kiwi/Aussie Rock
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/exchange/3"
          onClick={[Function]}
        >
          Bark bark woof
        </a>
      </li>
    </ul>
  </section>
  <section
    className="all-exchanges dashboard-section"
  >
    <h2>
      All Exchanges
    </h2>
    <ul>
      <li>
        <a
          aria-current={null}
          href="/exchange/1"
          onClick={[Function]}
        >
          The Roots
        </a>
        <p
          className="date-created"
        >
          2/18/2021
        </p>
      </li>
      <li>
        <a
          aria-current={null}
          href="/exchange/2"
          onClick={[Function]}
        >
          Kiwi/Aussie Rock
        </a>
        <p
          className="date-created"
        >
          2/18/2021
        </p>
      </li>
      <li>
        <a
          aria-current={null}
          href="/exchange/3"
          onClick={[Function]}
        >
          Bark bark woof
        </a>
        <p
          className="date-created"
        >
          2/18/2021
        </p>
      </li>
      <li>
        <a
          aria-current={null}
          href="/exchange/4"
          onClick={[Function]}
        >
          My influences
        </a>
        <p
          className="date-created"
        >
          2/18/2021
        </p>
      </li>
      <li>
        <a
          aria-current={null}
          href="/exchange/5"
          onClick={[Function]}
        >
          Wedding songs
        </a>
        <p
          className="date-created"
        >
          2/18/2021
        </p>
      </li>
    </ul>
  </section>
</main>
`;
